packages/cli/test/integration.js

```diff
@@ -378,6 +393,90 @@ test('login', async t => {
   t.is(auth.token, token);
 });
 
+test('default command should deploy directory', async t => {
+  const projectDir = fixture('deploy-default-with-sub-directory');
+  const target = 'output';
+
+  await vcLink(t, path.join(projectDir, target));
+
+  const { exitCode, stderr, stdout } = await execa(
+    binaryPath,
+    [
+      // omit the default "deploy" command
+      target,
+      ...defaultArgs,
+    ],
+    {
+      cwd: projectDir,
+    }
+  );
+
+  t.is(exitCode, 0, formatOutput({ stdout, stderr }));
+  t.regex(stdout, /https:\/\/output-.+\.vercel\.app/);
+  t.regex(
+    stderr || '',
+    /calling `vc` without a command or the `--cwd` option is deprecated, use `vc --cwd directory` or `vc deploy` instead/
+  );
+});
+
+test('default command should warn when deploying with conflicting subdirectory', async t => {
+  const projectDir = fixture('deploy-default-with-conflicting-sub-directory');
+  const target = 'list'; // command that conflicts with a sub directory
+
+  await vcLink(t, projectDir);
+
+  const { exitCode, stderr, stdout } = await execa(
+    binaryPath,
+    [
+      // omit the default "deploy" command
+      target,
+      ...defaultArgs,
+    ],
+    {
+      cwd: projectDir,
+    }
+  );
+
+  t.is(exitCode, 0, formatOutput({ stdout, stderr }));
+  t.regex(
+    stderr || '',
+    /Did you mean to deploy the subdirectory "list"\? Use `vc --cwd list` instead./
+  );
+
+  const listHeader = /project +latest deployment +state +age +username/;
+  t.regex(stdout || '', listHeader); // ensure `list` command still ran
+});
+
+test('default command should work with --cwd option', async t => {
+  const projectDir = fixture('deploy-default-with-conflicting-sub-directory');
+  const target = 'list'; // command that conflicts with a sub directory
+
+  await vcLink(t, path.join(projectDir, 'list'));
+
+  const { exitCode, stderr, stdout } = await execa(
+    binaryPath,
+    [
+      // omit the default "deploy" command
+      '--cwd',
+      target,
+      ...defaultArgs,
+    ],
+    {
+      cwd: projectDir,
+    }
+  );
+
+  t.is(exitCode, 0, formatOutput({ stderr, stdout }));
```

Comment:
Lets also test `stderr` here

Reply:
`stderr` is used for a bunch of non-error logs. Here's an example from this test:

```
Vercel CLI 24.0.1
Retrieving project…
Deploying endangeredmassa/list
Inspect: https://vercel.com/endangeredmassa/list/6BP3c8mUz536cMPcW8RecTrN3Evh [862ms]
Queued
Building
Completing
Preview: https://list-endangeredmassa.vercel.app [copied to clipboard] [5s]
To deploy to production (list-rosy.vercel.app), run `vercel --prod`
```

I added an assertion that the warning doesn't show up, at least.

Reply:
Maybe I'm missing it. Which line is the assertion? Did you forget to git push?

Reply:
Sorry, I added a `notRegex` for this, but a later change removed it. We're no longer showing that warning at all, which makes testing for it not showing up here useless.

Now stderr will only have the example I pasted above. Is there anything useful we can assert here given the latest state of the code?